### DISCUSSION TITLE
How to limit the uploaded file size and type?

### DISCUSSION THREAD

**Discussion by canimu** [directus/directus#7354] [Archive]
Hey, guys.
In order to manage the server capacity and protect the security, I need to limit the file's size and type before they are uploaded to the server(ex: maximum 30MB per file or only approve PNG file).
In the first step, I tried to use Collection validation while creating in permission page like below :

```json
{
   "filesize":{
       "_lt":30000}
}
```

or 

```json
{
    "file_name":{
         "_contains":".PNG"}
}
```

After I set it up, the big-size or incorrect type file can still be uploaded, so it seems doesn't work.
And then, I searched for some solutions in hook pages, but file.upload hook hasn't support .before modifier yet.
Is there any other solution? Thanks for helping!

**Comment by rijkvanzanten** [comment]
> I tried to use Collection validation while creating in permission page like below :

Where did you configure this?

**Reply by canimu** [reply]
I've created a role called Author. And I tried to limit in Author's permission like below:
(In author -> Directus_file -> filed validation.)
![image](https://user-images.githubusercontent.com/12815245/129203663-f5223c94-ba9c-4dc1-a03f-3a13e23af27e.png)

**Reply by Nierrrrrrr** [reply]
I have the same requirement and the same setting to Field Validation in "Directus Files" - "Create" permission and it doesn't work as expected.

**Comment by tiaan-va** [comment]
It has been nearly 5 years since this was posted, and it still remains the case today. The validation for filesize when uploading files simple does not work at all which is very annoying when Directus boast to have such good granular control over user permissions. Can someone tell me why validation for filesize does not work even in the slightest?
